pkg/3scale/amp/component/backend.go

```diff
@@ -52,6 +53,10 @@ func (backend *Backend) Objects() []common.KubernetesObject {
 	redisSecret := backend.RedisSecret()
 	listenerSecret := backend.ListenerSecret()
 
+	workerPDB := backend.WorkerPodDisruptionBudget()
+	cronPDB := backend.CronPodDisruptionBudget()
+	listenerPDB := backend.ListenerPodDisruptionBudget()
```

Comment:
I see the PodDisruptionBudget objects have been added into the "Objects" methods of every component that uses them.
The "Objects" methods in the components are used by the templates code to generate the templates.

Just to clarify, in this repository we maintain two methods of deploying a 3scale API Management solution:
* The 3scale-operator, via the APIManager resource
* The OpenShift templates, that are autogenerated from source code (https://github.com/3scale/3scale-operator/blob/master/doc/development.md#building-3scale-templates)

The "Objects" methods are only used by the templates generation code at this moment. Just as a thought for possible refactorings/improvements not related to this PR, this makes me thing that maybe this methods shouldn't belong to the "component" package and probably should belong to the templates-related code only.

With the current implementation, PodDisruptionBudgets are added in all template files. This is a difference in behaviour compared to the operator.

I think we have the following possible approaches:
* Make this functionality an operator functionality only. Not include the PodDisruptionBudgets in any template (not even the HA one). To do this it would be as simple as not including the PodDisruptionBudgets objects in the "Objects" methods
* Make this functionality only available in the HA template. The best solution I can think to this is, in the templates code somehow provide a flag to make the PodDisruptionBudgets instantiated depending on the template being used. As it is done right now, this means that the component code (not template-specific code) might probably have to have a new attribute to do this, which means impact in both templates and operator code. Maybe a better solution in this case would be move the "Objects" methods from the component package to the templates one. This way probably we would not need to change code in the component package.
* Make this functionality available in all templates. As I explained, this is somehow inconsistent between how operators and templates would work, as for example the standard template (non-ha) would have a PodDisruptionBudget but the standard APIManager deployment with operator wouldn't.

What do you think @eguzki?

Reply:
I am happy with whatever approach you guys suggested here. Another approach I think could work is to have a separate yaml file just for the pdbs. E.g. `amp-pdbs.yaml`. This way users can decide if they want to create PodDiscruptionBudgets by applying this file.

Reply:
We should make operator and templates be as consistent as we can. That means, try to keep the differences when deploying OCP resources using templates and the operator to the minimum.

Reply:
"HA template" is a confusing name. That template should be called "External Databases" template. 

I think we should only allow PDB for deployments recommended for production use, i.e. "External Databases" templates and when "HA" mode is enabled in APImanager CR

Reply:
@eguzki completely agree. I am happy to remove this from the templates for now if you guys prefer.

Reply:
@eguzki Looks like we both commented at the same time. So looks like you guys prefer to only have PDBs in the `HA` template? If so I am happy to make the change.

Reply:
@eguzki @miguelsorianod Per our discussion, I have updated the code to only generate PDBs for the HA template. Do you mind reviewing it again and see if you guys are happy with the approach I took. Thanks.

Reply:
Hi,

We are happy with the approach taken :+1: . We think adding configurability to the template adapters is a goo approach. In fact, we followed the same approach for a similar use case one PR that has not been merged yet (https://github.com/3scale/3scale-operator/pull/295).

Reply:
HA template implementation looks good to me :)